src/cli/cli.cpp

```diff
@@ -2058,15 +2058,45 @@ otError Interpreter::ProcessIpAddrDel(Arg aArgs[])
 
 otError Interpreter::ProcessIpAddr(Arg aArgs[])
 {
-    otError error = OT_ERROR_NONE;
+    otError error   = OT_ERROR_NONE;
+    bool    verbose = false;
+
+    if (aArgs[0] == "-v")
+    {
+        aArgs++;
+        verbose = true;
+    }
 
     if (aArgs[0].IsEmpty())
     {
         const otNetifAddress *unicastAddrs = otIp6GetUnicastAddresses(GetInstancePtr());
 
         for (const otNetifAddress *addr = unicastAddrs; addr; addr = addr->mNext)
         {
-            OutputIp6AddressLine(addr->mAddress);
+            OutputIp6Address(addr->mAddress);
+            if (verbose)
+            {
+                OutputFormat(" origin:");
+                switch (addr->mAddressOrigin)
```

Comment:
How about following this design pattern for mapping enum to string:
https://github.com/openthread/openthread/blob/ca6cbba4724e7560441fcdcf6a3fb3c56e0b0043/src/cli/cli.cpp#L1762-L1772